src/Mvc/Mvc.Core/src/ModelBinding/Binders/ComplexObjectModelBinderProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// An <see cref="IModelBinderProvider"/> for complex types.
+    /// </summary>
+    public class ComplexObjectModelBinderProvider : IModelBinderProvider
+    {
+        /// <inheritdoc />
+        public IModelBinder GetBinder(ModelBinderProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var metadata = context.Metadata;
+            if (metadata.IsComplexType && !metadata.IsCollectionType)
+            {
+                var loggerFactory = context.Services.GetRequiredService<ILoggerFactory>();
+                var parameterBinders = GetParameterBinders(context);
+
+                var propertyBinders = new Dictionary<ModelMetadata, IModelBinder>();
+                for (var i = 0; i < context.Metadata.Properties.Count; i++)
+                {
+                    var property = context.Metadata.Properties[i];
+                    propertyBinders.Add(property, context.CreateBinder(property));
+                }
+
+                return new ComplexObjectModelBinder(propertyBinders, parameterBinders, loggerFactory.CreateLogger<ComplexObjectModelBinder>());
```

Comment:
nit: No need to inline logger creation; suggest creating the `ILogger` just after getting the `loggerFactory`.